actionlint doesn't work well if shellcheck is installed by Lazy Install

## aqua version

Please use the latest version.

```console
$ aqua -v
aqua version 0.10.0 (3338a2ec9fbb63f6a5969cea48d6bb376f3b4a95)
```

## Environment

:warning: aqua doesn't support Windows.

* OS (Linux, macOS, etc): Linux and macOS
* CPU Architecture (amd64, arm64, etc): amd64

## Overview

actionlint internally executes shellcheck.
The output of actionlint is strange when there are multiple `run` steps and shellcheck is installed by lazy install.

## How to reproduce

* aqua.yaml
* executed command and output

> aqua.yaml should be not partial but complete configuration.
> Please remove unneeded configuration to reproduce the issue.

```yaml
# aqua.yaml
registries:
- type: standard
  ref: v0.13.1 # renovate: depName=aquaproj/aqua-registry

packages:
- name: koalaman/shellcheck@v0.8.0
- name: rhysd/actionlint@v1.6.8
```

.github/workflows/actionlint.yaml

```yaml
---
name: actionlint
on:
  pull_request:
    branches: [master]
jobs:
  default:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: aquaproj/aqua-installer@v0.6.0
      with:
        aqua_version: v0.10.0
    - run: echo ~/.aqua/bin
    - run: actionlint
```

```console
$ aqua i -l
$ rm -R /Users/shunsuke-suzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/koalaman/shellcheck/v0.8.0
$ actionlint
```

## Debug Output

Please set the environment variable `AQUA_LOG_LEVEL=debug` and run aqua and tell us the debug output.

```console
$ AQUA_LOG_LEVEL=debug actionlint
DEBU[0000] CLI args                                      aqua_version=0.10.0 config= log_level=debug program=aqua
DEBU[0000] install the package                           aqua_version=0.10.0 package_name=rhysd/actionlint package_version=v1.6.8 program=aqua registry=standard
DEBU[0000] check if the package is already installed     aqua_version=0.10.0 package_name=rhysd/actionlint package_version=v1.6.8 program=aqua registry=standard
DEBU[0000] check the permission                          aqua_version=0.10.0 file_name=actionlint program=aqua
DEBU[0000] execute the command                           aqua_version=0.10.0 exe_path=/Users/shunsuke-suzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/rhysd/actionlint/v1.6.8/actionlint_1.6.8_darwin_amd64.tar.gz/actionlint program=aqua
`/Users/shunsuke-suzuki/.local/share/aquaproj-aqua/bin/shellcheck --norc -f json -x --shell bash -e SC1091,SC2194,SC2050,SC2154 -` did not run successfully while checking script at line:15,col:7: /Users/shunsuke-suzuki/.local/share/aquaproj-aqua/bin/shellcheck exited with status 255 but stdout was empty. stderr: "time=\"2022-01-12T12:33:31+09:00\" level=debug msg=\"CLI args\" aqua_version=0.10.0 config= log_level=debug program=aqua\ntime=\"2022-01-12T12:33:31+09:00\" level=debug msg=\"install the package\" aqua_version=0.10.0 package_name=koalaman/shellcheck package_version=v0.8.0 program=aqua registry=standard\ntime=\"2022-01-12T12:33:31+09:00\" level=debug msg=\"check if the package is already installed\" aqua_version=0.10.0 package_name=koalaman/shellcheck package_version=v0.8.0 program=aqua registry=standard\ntime=\"2022-01-12T12:33:31+09:00\" level=info msg=\"download and unarchive the package\" aqua_version=0.10.0 package_name=koalaman/shellcheck package_version=v0.8.0 program=aqua registry=standard\ntime=\"2022-01-12T12:33:32+09:00\" level=info msg=\"retry installing the package\" aqua_version=0.10.0 error=\"reading file in tar archive: file already exists: /Users/shunsuke-suzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/koalaman/shellcheck/v0.8.0/shellcheck-v0.8.0.darwin.x86_64.tar.xz/shellcheck-v0.8.0/LICENSE.txt\" package_name=koalaman/shellcheck package_version=v0.8.0 program=aqua registry=standard retry_count=1\ntime=\"2022-01-12T12:33:32+09:00\" level=debug msg=\"check if the package is already installed\" aqua_version=0.10.0 package_name=koalaman/shellcheck package_version=v0.8.0 program=aqua registry=standard\ntime=\"2022-01-12T12:33:32+09:00\" level=debug msg=\"check the permission\" aqua_version=0.10.0 file_name=shellcheck program=aqua\ntime=\"2022-01-12T12:33:32+09:00\" level=debug msg=\"execute the command\" aqua_version=0.10.0 exe_path=/Users/shunsuke-suzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/koalaman/shellcheck/v0.8.0/shellcheck-v0.8.0.darwin.x86_64.tar.xz/shellcheck-v0.8.0/shellcheck program=aqua\ntime=\"2022-01-12T12:33:32+09:00\" level=debug msg=\"command was executed but it failed\" aqua_version=0.10.0 error=\"signal: killed\" exe_path=/Users/shunsuke-suzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/koalaman/shellcheck/v0.8.0/shellcheck-v0.8.0.darwin.x86_64.tar.xz/shellcheck-v0.8.0/shellcheck exit_code=-1 program=aqua\ntime=\"2022-01-12T12:33:32+09:00\" level=debug msg=\"command failed\" aqua_version=0.10.0 error=\"signal: killed\" exit_code=-1 program=aqua\n"
DEBU[0002] command was executed but it failed            aqua_version=0.10.0 error="exit status 3" exe_path=/Users/shunsuke-suzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/rhysd/actionlint/v1.6.8/actionlint_1.6.8_darwin_amd64.tar.gz/actionlint exit_code=3 program=aqua
DEBU[0002] command failed                                aqua_version=0.10.0 error="exit status 3" exit_code=3 program=aqua
```

## Expected Behaviour

```console
$ actionlint
```

## Actual Behaviour

```console
$ actionlint                                                                                                        
`/Users/shunsuke-suzuki/.local/share/aquaproj-aqua/bin/shellcheck --norc -f json -x --shell bash -e SC1091,SC2194,SC2050,SC2154 -` did not run successfully while checking script at line:15,col:7: /Users/shunsuke-suzuki/.local/share/aquaproj-aqua/bin/shellcheck exited with status 255 but stdout was empty. stderr: "time=\"2022-01-12T12:26:18+09:00\" level=info msg=\"download and unarchive the package\" aqua_version=0.10.0 package_name=koalaman/shellcheck package_version=v0.8.0 program=aqua registry=standard\ntime=\"2022-01-12T12:26:20+09:00\" level=info msg=\"retry installing the package\" aqua_version=0.10.0 error=\"reading file in tar archive: file already exists: /Users/shunsuke-suzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/koalaman/shellcheck/v0.8.0/shellcheck-v0.8.0.darwin.x86_64.tar.xz/shellcheck-v0.8.0/LICENSE.txt\" package_name=koalaman/shellcheck package_version=v0.8.0 program=aqua registry=standard retry_count=1\n"
```

## Important Factoids

```console
$ AQUA_LOG_LEVEL=error actionlint
`/Users/shunsuke-suzuki/.local/share/aquaproj-aqua/bin/shellcheck --norc -f json -x --shell bash -e SC1091,SC2194,SC2050,SC2154 -` did not run successfully while checking script at line:15,col:7: /Users/shunsuke-suzuki/.local/share/aquaproj-aqua/bin/shellcheck exited with status 255 but stdout was empty. stderr: ""
```

If actionlint detects no violation, this problem doesn't occur.
Even if actionlint isn't installed by aqua, this problem occurs.

```
$ which actionlint
/usr/local/bin/actionlint

$ actionlint -version
1.6.8
installed by building from source
built with go1.17.2 compiler for darwin/amd64

$ actionlint
`/Users/shunsuke-suzuki/.local/share/aquaproj-aqua/bin/shellcheck --norc -f json -x --shell bash -e SC1091,SC2194,SC2050,SC2154 -` did not run successfully while checking script at line:15,col:7: /Users/shunsuke-suzuki/.local/share/aquaproj-aqua/bin/shellcheck exited with status 255 but stdout was empty. stderr: "time=\"2022-01-12T12:37:10+09:00\" level=info msg=\"download and unarchive the package\" aqua_version=0.10.0 package_name=koalaman/shellcheck package_version=v0.8.0 program=aqua registry=standard\ntime=\"2022-01-12T12:37:11+09:00\" level=info msg=\"retry installing the package\" aqua_version=0.10.0 error=\"reading file in tar archive: file already exists: /Users/shunsuke-suzuki/.local/share/aquaproj-aqua/pkgs/github_release/github.com/koalaman/shellcheck/v0.8.0/shellcheck-v0.8.0.darwin.x86_64.tar.xz/shellcheck-v0.8.0/LICENSE.txt\" package_name=koalaman/shellcheck package_version=v0.8.0 program=aqua registry=standard retry_count=1\n"
```

## Workaround

Install shellcheck before executing actionlint.

```console
$ shellcheck -version # shellcheck is installed by Lazy Install
$ actionlint
```

## References

* https://github.com/rhysd/actionlint
* https://github.com/koalaman/shellcheck

--- a/pkg/controller/exec.go
+++ b/pkg/controller/exec.go
@@ -6,6 +6,7 @@ import (
 	"os"
 	"os/exec"
 	"strings"
+	"time"
 
 	"github.com/sirupsen/logrus"
 	"github.com/suzuki-shunsuke/go-error-with-exit-code/ecerror"
@@ -30,12 +31,40 @@ func (ctrl *Controller) Exec(ctx context.Context, param *Param, exeName string, 
 	if err != nil {
 		return err
 	}
-	if which.Package != nil {
+	if which.Package != nil { //nolint:nestif
+		logE := ctrl.logE().WithFields(logrus.Fields{
+			"exe_path": which.ExePath,
+			"package":  which.Package.Name,
+		})
 		if err := ctrl.installPackage(ctx, which.PkgInfo, which.Package, false); err != nil {
 			return err
 		}
+		for i := 0; i < 10; i++ {
+			logE.Debug("check if exec file exists")
+			if fi, err := os.Stat(which.ExePath); err == nil {
+				if isOwnerExecutable(fi.Mode()) {
+					break
+				}
+			}
+			logE.WithFields(logrus.Fields{
+				"retry_count": i + 1,
+			}).Debug("command isn't found. wait for lazy install")
+			if err := wait(ctx, 10*time.Millisecond); err != nil { //nolint:gomnd
+				return err
+			}
+		}
 	}
 	return ctrl.execCommand(ctx, which.ExePath, args)
+}
+
+func wait(ctx context.Context, duration time.Duration) error {
+	timer := time.NewTimer(duration)
+	select {
+	case <-timer.C:
+		return nil
+	case <-ctx.Done():
+		return ctx.Err() //nolint:wrapcheck
+	}
 }
 
 func (ctrl *Controller) findExecFileFromPkg(registries map[string]*RegistryContent, exeName string, pkg *Package) (*PackageInfo, *File) {
@@ -109,18 +138,31 @@ func (ctrl *Controller) findExecFile(ctx context.Context, cfgFilePath, exeName s
 }
 
 func (ctrl *Controller) execCommand(ctx context.Context, exePath string, args []string) error {
-	cmd := exec.Command(exePath, args...)
-	cmd.Stdin = ctrl.Stdin
-	cmd.Stdout = ctrl.Stdout
-	cmd.Stderr = ctrl.Stderr
-	runner := timeout.NewRunner(0)
-
 	logE := ctrl.logE().WithField("exe_path", exePath)
 	logE.Debug("execute the command")
-	if err := runner.Run(ctx, cmd); err != nil {
-		exitCode := cmd.ProcessState.ExitCode()
-		logerr.WithError(logE, err).WithField("exit_code", exitCode).Debug("command was executed but it failed")
-		return ecerror.Wrap(err, exitCode)
+	for i := 0; i < 10; i++ {
+		logE.Debug("execute the command")
+		cmd := exec.Command(exePath, args...)
+		cmd.Stdin = ctrl.Stdin
+		cmd.Stdout = ctrl.Stdout
+		cmd.Stderr = ctrl.Stderr
+		runner := timeout.NewRunner(0)
+		if err := runner.Run(ctx, cmd); err != nil {
+			exitCode := cmd.ProcessState.ExitCode()
+			// https://pkg.go.dev/os#ProcessState.ExitCode
+			// > ExitCode returns the exit code of the exited process,
+			// > or -1 if the process hasn't exited or was terminated by a signal.
+			if exitCode == -1 && ctx.Err() == nil {
+				logE.WithField("retry_count", i+1).Debug("the process isn't started. retry")
+				if err := wait(ctx, 10*time.Millisecond); err != nil { //nolint:gomnd
+					return err
+				}
+				continue
+			}
+			logerr.WithError(logE, err).WithField("exit_code", exitCode).Debug("command was executed but it failed")
+			return ecerror.Wrap(err, exitCode)
+		}
+		return nil
 	}
 	return nil
 }

